modules/exploits/windows/local/unquoted_service_path.rb

```diff
@@ -95,16 +95,20 @@ def check
 
   def generate_folders(fpath, quick)
     potential_paths = []
+    checked_paths = []
     fpath.reverse.each do |x|
       path = fpath[0..fpath.index(x)].join(' ')
-      path_no_file = path.split('\\')[0...-1].join('\\')
-      vprint_status("Checking writability to: #{path_no_file}")
       # when we test writability, we drop off last part since thats the file name
+      path_no_file = path.split('\\')[0...-1].join('\\')
+
+      next if checked_paths.include? path_no_file
+
+      checked_paths << path_no_file
       unless writable?(path_no_file)
-        vprint_error('Path not writable')
+        vprint_error("#{path_no_file} not writable")
```

Comment:
nit: Maybe says `"#{path_no_file} is not writable"`, or change the next message to `"#{path_no_file} writable"`, for consistency's sake.